lib/spack/spack/directory_layout.py

```diff
@@ -223,19 +228,24 @@ def spec_file_path(self, spec):
         return os.path.join(self.metadata_path(spec), self.spec_file_name)
 
     def metadata_path(self, spec):
-        return os.path.join(self.path_for_spec(spec), self.metadata_dir)
+        if self.check_upstream and spec.package.installed_upstream:
+            # TODO: This assumes that older spack versions use the same
```

Comment:
The metadata dir is configurable on the YamlDirectoryLayout object. I think we should remove that option (no code appears to use it) as part of this PR since we depend on it not being used in the future.